src/awkward/_v2/contents/content.py

```diff
@@ -519,10 +519,19 @@ def _getitem(self, where):
                 None,
                 self._nplike,
             )
-
             out = next._getitem_next(nextwhere[0], nextwhere[1:], None)
 
             if out.length == 0:
+                if isinstance(items[0], slice):
+                    if items[0].start is None or items[0].stop is None:
+                        return ak._v2.contents.RegularArray(
+                            out,
+                            0,
+                            out.size,
+                            None,
+                            None,
+                            self._nplike,
+                        )
```

Comment:
The first issue with generality (described above) can be fixed like this:

```suggestion
                if isinstance(items[0], slice):
                    return ak._v2.contents.RegularArray(
                        out,
                        0,
                        out.size,
                        None,
                        None,
                        self._nplike,
                    )
```